tiledb/cc/group.cc

```diff
@@ -22,16 +22,14 @@ void put_metadata_numpy(Group &group, const std::string &key, py::array value) {
     throw py::type_error(e.what());
   }
 
-  if (is_tdb_str(value_type) && value.size() > 1)
```

Comment:
We can't serialize this.

Reply:
After adding three tests, arrays of strings weren't handled the same way as for Array metadata.

These cases were supposed to be caught here: https://github.com/TileDB-Inc/TileDB-Py/blob/0be88d46010dd8f4f2087081b4d83a6acb48bb16/tiledb/cc/group.cc#L30-L32

`get_ncells` didn’t exhibit the same behavior as its Python counterpart: https://github.com/TileDB-Inc/TileDB-Py/blob/0be88d46010dd8f4f2087081b4d83a6acb48bb16/tiledb/datatypes.py#L47-L55

After updating `get_ncells`, only single-character numpy arrays are now allowed, similar to the handling of Array metadata.